### ISSUE TITLE
Font Library uploader apparently crashes with broken font files

### ISSUE THREAD

**Issue by peterwilsoncc** [WordPress/gutenberg#59578]
### Description

When a user attempts to upload a font file that is broken, the font library uploader tab apparently crashes.

Inspecting the console, an error is thrown but fails to be caught.

### Step-by-step reproduction instructions

MacOS instructions

1. Remove the first 100 bytes from a font file: `tail -c +100 OpenSans-Bold.ttf > _broken-font.ttf`
2. Open the Font Library uploader tab via the site editor
3. Attempt to upload the broken font file
4. The uploader will show the upload progress bar for the foreseeable future

### Screenshots, screen recording, code snippet

![crash-uploader](https://github.com/WordPress/gutenberg/assets/519727/1be60578-dd71-4729-b2ef-68c26bfa47bb)

### Environment info

- WordPress 6.4
- Gutenberg trunk @ 619f0a3ea612f5bc836eafb6d3cfd46175bf1d8c

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by t-hamano** [comment]
I was also able to reproduce this problem. The console log output in debug mode is below.

```
lib-font.browser.js:810 Uncaught (in promise) Error: broken-font.ttf is either an unsupported font format, or not a font at all.
    at Font.fromDataBuffer (lib-font.browser.js:810:1)
    at getFontFaceMetadata (upload-fonts.js:110:11)
    at async upload-fonts.js:85:26
    at async Promise.all (:8888/wp-admin/index 0)
    at async loadFiles (upload-fonts.js:83:27)
```

I'll attach the broken font so that anyone can reproduce this problem. This is a font with the first 100 bytes of the Roboto Regular font removed.

[broken-font.zip](https://github.com/WordPress/gutenberg/files/14490210/broken-font.zip)

**Comment by t-hamano** [comment]
I think the problem is that the `fromDataBuffer` method throws an exception if the font file is incorrect, but that exception is not caught.

https://github.com/WordPress/gutenberg/blob/64fb8dd1e99bcebe3790c310d4fd918001d8939d/packages/edit-site/lib/lib-font.browser.js#L809-L813

https://github.com/WordPress/gutenberg/blob/64fb8dd1e99bcebe3790c310d4fd918001d8939d/packages/edit-site/src/components/global-styles/font-library-modal/upload-fonts.js#L110-L114

cc @matiasbenedetto @pbking

**Comment by youknowriad** [comment]
Since RC1 is today, this probably won't be fixed for 6.5. So I'm removing from the board.
